buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -267,6 +267,7 @@ public ByteBuf ensureWritable(int minWritableBytes) {
     }
 
     private void ensureWritable0(int minWritableBytes) {
+        ensureAccessible();
```

Comment:
this feels backwards, isn't it usually the public method eg. mymethod invoking the private mymethod0?